docs/Webhooks.md

```diff
@@ -1,179 +1,97 @@
 
+The Waypoint service provides a specialized Server-Sent Events (SSE) endpoint for cases where you need to wait for a specific
+event that you know is coming. This is particularly useful when you need to track the state change of a specific entity
+and have all the necessary filter information.
 
-async def main():
-    # Create a client and subscribe to topics
-    topics = ["connections", "credentials", "proofs", "endorsements", "basic-messages"]
-    client = PubSubClient([*topics], callback=on_events)
+The Waypoint service exposes a single SSE endpoint, via the main app, for event streaming:
 
-    client.start_client(f"ws://webhooks.cloudapi.127.0.0.1.nip.io/pubsub")
-    await client.wait_until_done()
+- `GET` `/v1/sse/{wallet_id}/{topic}/{field}/{field_id}/{desired_state}`
 
+This endpoint is designed for targeted event retrieval where you can specify exact filters to wait for a particular event.
 
-asyncio.run(main())
-```
+### How to use
 
-All webhooks are by default logged to the container's stdout, which can be handy
-for debugging or development. You can easily pipe them into a file. On a Unix
-machine, you can use this command to follow the webhook logs:
+The endpoint requires specific filter parameters:
 
-```bash
-kubectl logs -f $(kubectl get pods \
-  -l app.kubernetes.io/instance=governance-webhooks-web \
-  -o jsonpath="{.items[0].metadata.name}")
-```
+- `wallet_id`: Identifier for the specific wallet
+- `topic`: The event topic to subscribe to
+- `field`: Filter field from the event payload
+- `field_id`: Specific value for the filter field
+- `desired_state`: The desired state of the event
 
-### Non-Python options
+For example, if you're waiting for a specific connection to reach the 'completed' state, you would use:
 
-Listening to webhooks the subscriber way (not long polling via HTTP) is not
-limited to the Python example given above. In fact, all one needs is a
-websocket-based RPC client.
+- `field`: "connection_id"
+- `field_id`: "<your_connection_id>"
+- `desired_state`: "completed"
 
-Here are two examples:
-
-```bash
-websocat --exit-on-eof --text \
-  ws://webhooks.cloudapi.127.0.0.1.nip.io/pubsub \
-  exec:'{"request": {"method": "subscribe", "arguments": {"topics": ["proofs", "endorsements", "oob", "out_of_band", "connections", "basic-messages", "credentials"]}}}'
-```
+The stream will remain open until the desired event (matching the filters) is found and returned, at which point the stream
+will be closed.
 
-or
+Valid topics include:
 
-```bash
-wscat -c ws://webhooks.cloudapi.127.0.0.1.nip.io \
-  -x '{"request": {"method": "subscribe", "arguments": {"topics": ["proofs", "endorsements", "oob", "out_of_band", "connections", "basic-messages", "credentials"]}}}' \
-  -w 99999
+```python
+topics = Literal[
+    "basic-messages",
+    "connections",
+    "proofs",
+    "credentials",
+    "endorsements",
+    "oob",
+    "revocation",
+    "issuer_cred_rev",
+    "problem_report",
+]
```

Comment:
Let's alphabetize these, and add: credentials_indy, credentials_ld

There's also `deleted_credential` but I think we may want to refactor that to be part of `credentials`. Let's see